Vinay - Let a user know through a popup that Permissions have been changed for them

vinayv-456 opened the pull request:
# Description
If an account changes permissions for another account, then when the account which has permissions edited, logs in, there should be a popup letting them know that permissions have been edited in this account. Only should happen for the first time the user logs into account after permissions were edited.

## Related PRS (if any):
This frontend PR is related to the [#1336](https://github.com/OneCommunityGlobal/HGNRest/pull/1336) backend PR.

…

## Main changes explained:
- src/components/Header/Header.jsx: added popup to show update in permission. refactor popup component. handle on close of permission change popup 
- src/components/PopUpBar/PopUpBar.jsx: updated the PopupBar component to accept the message and reused the component to notify the change in permission
- src/components/PopUpBar/__tests__/PopUpBar.test.jsx: updated the test to support the above changes
…

## How to test:
1. check into "vinay/notify-user-new-permission" branch
2. This frontend PR is related to the [#1278](https://github.com/OneCommunityGlobal/HGNRest/pull/1278) backend PR.
3. Login using owner account->Other links -> Permission Management -> Manage User Permission -> Make some permission changes for another account and then submit to save changes.
4. Log into the account that has permissions edited and now there should be a popup like ‘Heads Up, there were permission changes made to this account ’ only for the first time.

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/a43f9ddb-b5d7-41b4-af7c-3d039593da86



## Note:
Include the information the reviewers need to know.

KH07 reviewed (approved):
Tested functioned as intended. I checked out the current branch, ran npm install, and started the app locally. Logging in with an owner account, changed permission for a volunteer, including one single change and multiple changes. Then logged in with the volunteer account, notice popup shows up at the top.
<img width="1558" alt="PR#3396 Notice" src="https://github.com/user-attachments/assets/acb2c64a-af3c-45c5-86f1-18ebd710b12c" />

JuliaHa0902 reviewed (changes requested):
Hi Vinay, 
I have checked out the branch and test updating the permission. The popup showed that there were permission changes made to this account as expected. 
![Screenshot 2025-04-21 182454](https://github.com/user-attachments/assets/ad8da299-4d41-49dc-be85-366a3a33b6e0)
However, I have a problem closing the popup. When I check the console, I got a 403 error when trying to close the popup.
![Screenshot 2025-04-21 184457](https://github.com/user-attachments/assets/b905d261-5559-4743-8020-a592d09eb24d)
Also, just a minor suggestion for design, the white text on top of a bright yellow background is very hard to see. I would suggest changing the text color to black.

rahulprasad2823 reviewed (commented):
Checked out to the branch and followed the provided steps. Logged in as the owner account, made permission changes for another user, and submitted successfully. Then logged into the updated account, but did not see the expected "Heads Up" popup notification regarding the permission changes.

https://github.com/user-attachments/assets/662bc1ec-6745-4e2c-87c4-80a3fe194bf2

vinayv-456 commented:
> Hi Vinay, I have checked out the branch and test updating the permission. The popup showed that there were permission changes made to this account as expected. ![Screenshot 2025-04-21 182454](https://private-user-images.githubusercontent.com/43044560/435849083-ad8da299-4d41-49dc-be85-366a3a33b6e0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUzNTQ4MDYsIm5iZiI6MTc0NTM1NDUwNiwicGF0aCI6Ii80MzA0NDU2MC80MzU4NDkwODMtYWQ4ZGEyOTktNGQ0MS00OWRjLWJlODUtMzY2YTNhMzNiNmUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIyVDIwNDE0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk3OGYwYTk1ODM5NTE4OWQ5ODRhNzA3NjQ5MWZhZWQyZTAzODU4NGE1ZGNiNDMzNjFlOTA3Yzg0OTgyYzczM2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nXU4jseGcg2yfDlRK7c6nS0yOLTQnCrgKuEYcBZY3GY) However, I have a problem closing the popup. When I check the console, I got a 403 error when trying to close the popup. ![Screenshot 2025-04-21 184457](https://private-user-images.githubusercontent.com/43044560/435849190-b905d261-5559-4743-8020-a592d09eb24d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUzNTQ4MDYsIm5iZiI6MTc0NTM1NDUwNiwicGF0aCI6Ii80MzA0NDU2MC80MzU4NDkxOTAtYjkwNWQyNjEtNTU1OS00NzQzLTgwMjAtYTU5MmQwOWViMjRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIyVDIwNDE0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4ZTA2ZDBlMDUzOGIwODNhZjk3YzM3OTQ5ZjQ0NTMwNDA3ZWU4ODYzMTE5YjIzNTIwYTlmMjAyMTAxMzBkMDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fjlvOCtXTRJzxzQbkO3_CaOz5QXbDyJZFkkeqmXNsfU) Also, just a minor suggestion for design, the white text on top of a bright yellow background is very hard to see. I would suggest changing the text color to black.

Hi @JuliaHa0902 ,
The above permission issue is comming if the users role is neither admin nor owner.  So checked with the volunteer role account.
There was unwanted teamcode permission check even during non updation of team code. This is fixed now.
 Even updated the text color to black for better visibility as you suggested.
please pull the latest frontend and backend code to check the fixes.

![image](https://github.com/user-attachments/assets/cc95f629-6e95-4f8c-bff2-9ce2a069b9a5)

JuliaHa0902 reviewed (changes requested):
Hi Vinay, I have checked out the branch and tested it again. The banner looks good now. The permission error has been fixed. Things work as expected.
![Screenshot 2025-04-22 180420](https://github.com/user-attachments/assets/76a119d2-d72d-4920-b230-beb0d8eb9393)
There is just a minor GUI problem with the mobile view. The text overlaps with the X icon. I would suggest adding some padding to the banner so it looks better in mobile or tablet view.
Also, closing the banner seems to take a little bit of time. If you can add a spinning wheel so user knows that the website is processing, that would be nice. Or you can hide the banner first before the API return.

gopikalakshmia reviewed (dismissed):
Checked out the branch, created a new user for testing, made some permission changes, logged into the new user's account, and verified that the heads-up message is shown on the screen. When logged in the second time, the heads-up message is not shown—works as intended.Approving the PR
<img width="468" alt="image" src="https://github.com/user-attachments/assets/8ff8018a-b249-4ce6-abc6-ae0dedf1d7ea" />

AnthonyWeathers reviewed (dismissed):
Checked out this branch along with the updated backend for this from HGNRest [PR#1336](https://github.com/OneCommunityGlobal/HGNRest/pull/1336). The pop up appeared and worked as expected with no issues on testing.

vinayv-456 commented:
> Hi Vinay, I have checked out the branch and tested it again. The banner looks good now. The permission error has been fixed. Things work as expected. ![Screenshot 2025-04-22 180420](https://private-user-images.githubusercontent.com/43044560/436299094-76a119d2-d72d-4920-b230-beb0d8eb9393.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYwMzQyMDYsIm5iZiI6MTc0NjAzMzkwNiwicGF0aCI6Ii80MzA0NDU2MC80MzYyOTkwOTQtNzZhMTE5ZDItZDcyZC00OTIwLWIyMzAtYmViMGQ4ZWI5MzkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDMwVDE3MjUwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlZjk5NTY4MmMxNGQ4YTc0MTYyYmMxOWM5NjllYWVmMjYyNGEyYzc1NWI4NjBjMGU4YjIwYTRlNGQ0MmZiOGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ajSjF8AUQYUuEwIJkVp6asy-yUEYrLf20ti4BKl-qso) There is just a minor GUI problem with the mobile view. The text overlaps with the X icon. I would suggest adding some padding to the banner so it looks better in mobile or tablet view. ![Screenshot 2025-04-22 180544](https://private-user-images.githubusercontent.com/43044560/436300568-f0cba6ae-139f-43ae-baee-74be53bd046c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYwMzQyMDYsIm5iZiI6MTc0NjAzMzkwNiwicGF0aCI6Ii80MzA0NDU2MC80MzYzMDA1NjgtZjBjYmE2YWUtMTM5Zi00M2FlLWJhZWUtNzRiZTUzYmQwNDZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDMwVDE3MjUwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRiZDkyYWZjNzUwNjA2MjA0MDc1ZjExZjBiNGI5ODZiYTEyYzYwMWY0Nzc1Mjc5NmNlZWQ3ODMyZDMxYzUzMmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zUQV5RwuCfIh1kgfw0IKabp6xHP7D5lbmqaMJJFlAP0)
> 
> Also, closing the banner seems to take a little bit of time. If you can add a spinning wheel so user knows that the website is processing, that would be nice. Or you can hide the banner first before the API return.

@JuliaHa0902 ,
Added loader and fixed padding. Please verify.

one-community commented:
Thank you all, merging!
